fog/view/server/src/router_admin_service.rs

```diff
@@ -44,8 +43,13 @@ impl FogViewRouterAdminService {
                 logger,
             )
         })?;
-        let mut shard_clients = self.shard_clients.write().expect("RwLock Poisoned");
-        if shard_clients.keys().contains(&view_store_uri) {
+        let mut shards = self.shards.write().expect("RwLock Poisoned");
+        if shards
+            .iter()
+            .map(|shard| shard.uri.clone())
+            .collect::<HashSet<FogViewStoreUri>>()
+            .contains(&view_store_uri)
```

Comment:
This seems like it should be expressable as `shards.iter().any(|shard| shard.uri.clone() == view_store_uri)` or something similar, without having to collect up a whole `HashSet`.

Reply:
Done.